### DISCUSSION TITLE
Is there a way to store data in the wasm?

### DISCUSSION THREAD

**Discussion by Juliiiian** [elringus/bootsharp#151] [Q&A]
I have a function Load that loads a world from a byte arr. Is there any way to store the loaded world and for example access the loaded information later through another function. Since only static functions are allowed, it's kinda hard to do it. (Sorry if this is a bit basic. I am just getting into C#)

```c# 
public WorldSerialization world = new WorldSerialization();

[JSEvent] // Used in JS as Program.Log.subscribe(..)
public static partial void Log(string message);

[JSInvokable] 
 public void Load(byte[] data) {
     world.Load(data);
     Log($"Loading has worked! Map size: {world.world.size}");
 }
```

```c# 

[JSEvent] // Used in JS as Program.Log.subscribe(..)
public static partial void Log(string message);

[JSInvokable] 
 public static void Load(byte[] data) {
     WorldSerialization world = new WorldSerialization();
     world.Load(data);
     Log($"Loading has worked! Map size: {world.world.size}");
 }
```

Even when making it static and just saving the world inside the function (which is not convenient) it will run, but not save the data into the world (I guess since the WorldSerialization is not static)

Thank you for your help!

**Comment by elringus** [comment]
Hi, Not sure what you mean by save. If it's about persisting the data, so it can be retrieved across, say, web page visits, you can use any of the available browser persistence models, such as IndexedDb or FileSystem Access API. Also, Bootsharp is not limited to static functions, you can use instances as well: https://bootsharp.com/guide/interop-instances.

**Reply by Juliiiian** [reply]
Yeah, my wording wasn't quite right. I want to load the map and after that is done decide on the action of the user to generate a Mesh from the loaded data (all in one browser tab, no need for FileSystem Access yet). And since I can't export the class that holds the whole data, I want to store it in a world variable inside the class. 
I looked through instances and the class still doesn't save/store the data inside the variable (maybe I missed something).  
I just need a persistent variable/property on the class between function calls.

**Reply by Juliiiian** [reply]
And one small thing I was curious about is, is there any way to specify the memory usage? Since the world can have up to 170MB, could there be any problem with such a big size?

**Reply by elringus** [reply]
I don't think any of that is related to Bootsharp specifically, it's all in the domain of C#/.NET.

**Reply by Juliiiian** [reply]
I am sorry if I don't understand it correctly, prob like you said a C#/.NET thing. 
But is there no way to (simple example) have a property on the class which stores a simple string. And then with one function I can set the string and with another I read/get it back.
